lib/rspec/expectations/failure_aggregator.rb

```diff
@@ -0,0 +1,159 @@
+module RSpec
+  module Expectations
+    # @private
+    class FailureAggregator
+      attr_reader :block_label, :metadata
+
+      def aggregate
+        RSpec::Support.with_failure_notifier(self) do
+          begin
+            yield
+          rescue ExpectationNotMetError => e
+            failures << e
+          rescue Exception => e
```

Comment:
nope, still don't get it. I guess this is supported by already-written code somewhere.